below/model/src/collector.rs

```diff
@@ -306,6 +308,17 @@ fn collect_sample(
                 }
             }
         },
+        tc: if !options.enable_tc_stats {
+            None
+        } else {
+            match tc::tc_stats() {
```

Comment:
How expensive is this since we're talking over netlink sockets and it's blocking? We generally try to make sure the main collection is fast to minimize collection skew and make sure we're still able to collect data under high system load.

More expensive collection can be made "best effort" and delegated to some other thread (e.g. via `AsyncCollectorPlugin`).

Reply:
I do not think netlink operations would be expensive since they directly communicate with the kernel. However, I believe the receive operation is blocking. It could be slow only if there are, say 1000+ interfaces, which I am not sure is likely.

That being said, I tried to duplicate the implementation for `gpu_stats` for `tc` in this [commit](https://github.com/facebookincubator/below/pull/8210/commits/83f218f44c758bc4ca7004a9e4335caf75fcc4a8). It seems to work but would appreciate a closer look.

Reply:
That commit looks good. Note that the async collection means there may be more skew. e.g. TC collected at t=0s but sample collected in main thread at t=4s.